### PR TITLE
FEAT: Move ibis.pandas.udf.reduction and ibis.pandas.udf.analytics to ibis.udf.vectorized namespace

### PR DESCRIPTION
**Pull request by icexelloss**
This PR refactor `ibis.pandas.udf.reduction` and `ibis.pandas.udf.analytics` into `ibis.udf.vectorized`.

This is similar to the first vectorized UDF PR:
https://github.com/ibis-project/ibis/pull/2047

### CODE REVIEW

**Comment by costrouc** [comment]
My PR https://github.com/ibis-project/ibis/pull/1997 is failing in the test suite in the same way. Do you know why? 

> isort....................................................................Failed
> - hook id: isort
> - files were modified by this hook

> Fixing ibis/impala/client.py
> Fixing ibis/impala/pandas_interop.py
> Fixing ibis/impala/tests/test_udf.py
> Fixing ibis/impala/tests/test_partition.py
> Fixing ibis/filesystems.py
> Fixing ibis/spark/tests/test_ddl.py
> Fixing ibis/tests/test_filesystems.py
> Fixing ibis/impala/tests/test_parquet_ddl.py
> Fixing ibis/impala/tests/test_ddl.py

**Comment by xmnlab** [comment]
@costrouc @icexelloss 
it seems something related to posixpath changed. I am updating the files here https://github.com/ibis-project/ibis/pull/2211

File: ibis/pandas/tests/test_udf.py [hunk 1]
```diff
@@ -98,10 +98,9 @@ def a_single_number(**kwargs):
 
 
 @udf.reduction(
-    input_type=[dt.double, dt.Array(dt.double)],
-    output_type=dt.Array(dt.double),
+    input_type=[dt.double], output_type=dt.Array(dt.double),
 )
-def quantiles(series, quantiles):
+def quantiles(series, *, quantiles):
```
Comment (jreback) [inline ibis/pandas/tests/test_udf.py:103]: do we require kwargs only?

File: ibis/pandas/udf.py [hunk 2]
```diff
@@ -211,130 +136,6 @@ def nullable(datatype):
...
-
-
-def valid_function_signature(input_type, func):
-    """Check that the declared number of inputs (the length of `input_type`)
-    and the number of inputs to `func` are equal.
-
-    Parameters
-    ----------
-    input_type : List[DataType]
-    func : callable
```
Comment (jreback) [inline ibis/pandas/udf.py:316]: so most of this was duplicated code?

Comment (icexelloss) [reply ibis/pandas/udf.py:316]: Yes these code are either moved to `ibis.udf.vectorized` or to `pre_execute_reduction_udf` in this file

File: ibis/pandas/tests/test_udf.py (hunk 1 above)
Comment (icexelloss) [reply ibis/pandas/tests/test_udf.py:103]: This is correct. I think this makes the API a little cleaner (what are data/col args and what are configuration args). Basically I want to only allow configuration args as keyword only args.

File: ibis/expr/operations.py [hunk 3]
```diff
@@ -3450,11 +3450,72 @@ class GeoAsText(GeoSpatialUnOp):
...
+        self._output_type = output_type
+
+    @property
+    def inputs(self):
+        return self.func_args
+
+    def output_type(self):
+        return self._output_type.scalar_type()
+
+    def root_tables(self):
```
Comment (jreback) [inline ibis/expr/operations.py:3504]: might make sense to have a BaseUDF class to inherit from (followon) if you think is worth it

File: ibis/pandas/tests/test_udf.py (hunk 1 above)
Comment (jreback) [reply ibis/pandas/tests/test_udf.py:103]: ok right but the actual interface doesn't require that correct? can you also test with positional args (e.g. lke the original), just for completeness. I agree using kwargs only is a good thing. maybe should enforce it.

File: ibis/expr/operations.py (hunk 3 above)
Comment (icexelloss) [reply ibis/expr/operations.py:3504]: Yeah I tried that approach first. But ibis note really dislike multi inheritance (complains about template conflicting in meta classes)

File: ibis/pandas/tests/test_udf.py (hunk 1 above)
Comment (icexelloss) [reply ibis/pandas/tests/test_udf.py:103]: Let me add a few test/doc to clarify what is allowed and not allowed

**Comment by jreback** [comment]
very nice @icexelloss

**Comment by icexelloss** [comment]
Thank you! @jreback

**Pull request merged**